File: inventory-service/src/integrationTest/java/br/com/maccommerce/inventoryservice/mock/DatabaseMock.java
package br.com.maccommerce.inventoryservice.mock;

import io.zonky.test.db.postgres.embedded.EmbeddedPostgres;

import java.io.IOException;

public class DatabaseMock {

    private final EmbeddedPostgres embeddedPostgres;

    public DatabaseMock() throws IOException {
        this.embeddedPostgres = EmbeddedPostgres.builder().setPort(5433).start();
    }

    public void stopServer() throws IOException {
        embeddedPostgres.close();
    }

}

File: product-service/src/integrationTest/kotlin/br/com/maccommerce/productservice/DatabaseMock.kt
package br.com.maccommerce.productservice

import br.com.maccommerce.productservice.domain.entity.Category
import br.com.maccommerce.productservice.domain.entity.Product
import br.com.maccommerce.productservice.domain.repository.CategoryRepository
import br.com.maccommerce.productservice.domain.repository.ProductRepository
import io.zonky.test.db.postgres.embedded.EmbeddedPostgres
import org.koin.core.KoinComponent
import org.koin.core.get

object DatabaseMock : KoinComponent {

    fun startServer(port: Int): EmbeddedPostgres = EmbeddedPostgres.builder().setPort(port).start()

    fun insertCategory(category: Category) = get<CategoryRepository>().persist(category)

    fun insertProduct(product: Product) = get<ProductRepository>().persist(product)

}